From 883b636e942148e8a6200dc661c7faa974216923 Mon Sep 17 00:00:00 2001
From: Dweep Advani <dweep.advani@broadcom.com>
Date: Mon, 20 May 2024 04:43:32 +0000
Subject: [PATCH 3/4] systemd: do not use -ftrivial-auto-var-init=zero

The -ftrivial-auto-var-init=zero was causing automatic variables to be
initialized to zero, which had huge performance impact on systemd
operations like mask/unmask/enable/disable. Do not build wih that flag.
Ref: https://gcc.gnu.org/pipermail/gcc-patches/2021-February/565514.html
Ref: https://github.com/systemd/systemd/pull/23297
---
 meson.build | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/meson.build b/meson.build
index 103febfd70..f5b1850750 100644
--- a/meson.build
+++ b/meson.build
@@ -469,13 +469,6 @@ if cc.get_id() == 'clang'
         ]
 endif
 
-if get_option('mode') == 'release'
-        # We could enable 'pattern' for developer mode, but that can interfere with
-        # valgrind and sanitizer builds. Also, clang does not zero-initialize unions,
-        # breaking some of our code (https://reviews.llvm.org/D68115).
-        possible_common_cc_flags += '-ftrivial-auto-var-init=zero'
-endif
-
 possible_cc_flags = [
         '-fno-strict-aliasing',
         '-fstrict-flex-arrays=1',
-- 
2.52.0

